src/plugins/URLIndicatorPlugin/URLIndicator.js

```diff
@@ -0,0 +1,109 @@
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define(
+    [],
+    function () {
+
+        // Set of connection states; changing among these states will be
+        // reflected in the indicator's appearance.
+        // CONNECTED: Everything nominal, expect to be able to read/write.
+        // DISCONNECTED: HTTP failed; maybe misconfigured, disconnected.
+        // PENDING: Still trying to connect, and haven't failed yet.
+        // ERROR: Got answer but it's due an internal server error or similar.
+        var CONNECTED = {
+                glyphClass: "ok"
+            },
+            PENDING = {
+                glyphClass: 'caution'
+            },
+            DISCONNECTED = {
+                glyphClass: "err"
+            };
+        function URLIndicator($http, $interval) {
+            var self = this;
+            // Get the current connection state and options provided
+            this.icon = "icon-" + (this.options.icon ? this.options.icon : "database");
+            this.URLpath = this.options.url;
+            this.label = this.options.label ? this.options.label : this.options.url;
+            this.interval = this.options.interval || 10000;
+            this.state = PENDING;
+
+            // Callback if the HTTP request to URLIndicatorPlugin fails
+            function handleError(e) {
+                self.state = DISCONNECTED;
+            }
+            // Callback if the HTTP request succeeds.
+            function handleResponse() {
+                self.state = CONNECTED;
+            }
+            // Try to connect to the URL, and update the indicator.
+            function updateIndicator() {
+                $http.get(self.URLpath).then(handleResponse, handleError);
+            }
+            // Update the indicator initially, and start polling.
+            updateIndicator();
+            $interval(updateIndicator, self.interval, 0, false);
+        }
+
+        URLIndicator.prototype.getCssClass = function () {
+            return this.icon;
+        };
+        URLIndicator.prototype.getGlyphClass = function () {
+            return this.state.glyphClass;
+        };
+        URLIndicator.prototype.getText = function () {
+            switch (this.state.glyphClass) {
+                case "ok": {
+                    return this.label + " is connected";
+                }
+                case "caution": {
+                    return "Checking status of " + this.label + " please stand by...";
+                }
+                case "err": {
+                    return this.label + " is offline";
+                }
+                default: {
+                    return "Checking status of " + this.label + " please stand by...";
```

Comment:
default clause should never occur (because state is always set in constructor, never unset), so remove it.